packages/core/src/types/redocly-yaml.ts

```diff
@@ -151,12 +152,338 @@ const RootConfigStyleguide: NodeType = {
+  required: ['adapters'],
+  additionalProperties: false,
+  properties: {
+    adapters: {
+      type: 'array',
+      items: devOnboardingAdapterConfigSchema,
+    },
+  },
+} as const;
+
+const responseHeaderSchema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    value: { type: 'string' },
+  },
+  additionalProperties: false,
+  required: ['name', 'value'],
+} as const;
+
+
+const apiConfigSchema = {
+  type: 'object',
+  additionalProperties: {
+    type: 'object',
+    properties: {
+      root: { type: 'string' },
+      rbac: { type: 'object', additionalProperties: true },
+      theme: {
+        type: 'object',
+        properties: {
+          openapi: { type: 'object', additionalProperties: true },
+        },
+        additionalProperties: false,
+      },
+      title: { type: 'string' },
+      metadata: { type: 'object', additionalProperties: true },
+    },
+    additionalProperties: true,
+    required: ['root'],
+  }
+} as const;
+
+const scorecardConfigSchema = {
+  type: 'object',
+  additionalProperties: true,
+  required: ['levels'],
+  properties: {
+    levels: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['name'],
+        properties: {
+          name: { type: 'string' },
+          extends: { type: 'array', items: { type: 'string' } },
+          rules: {
+            type: 'object',
+            additionalProperties: {
+              type: ['object', 'string'],
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    },
+    targets: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['where'],
+        properties: {
+          minimumLevel: { type: 'string' },
+          where: {
+            type: 'object',
+            required: ['metadata'],
+            properties: {
+              metadata: { type: 'object', additionalProperties: { type: 'string' } },
+            },
+            additionalProperties: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    },
+  },
+};
+
+const i18nConfigSchema = {
+  type: 'object',
+  properties: {
+    defaultLocale: {
+      type: 'string',
+    },
+    locales: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          code: {
+            type: 'string',
+          },
+          name: {
+            type: 'string',
+          },
+        },
+        required: ['code'],
+      },
+    },
+  },
+  required: ['defaultLocale', 'locales'],
+};
+
+const mockServerConfigSchema = {
+  type: 'object',
+  properties: {
+    off: { type: 'boolean', default: false },
+    position: { type: 'string', enum: ['first', 'last', 'replace', 'off'], default: 'first' },
+    strictExamples: { type: 'boolean', default: false },
+    errorIfForcedExampleNotFound: { type: 'boolean', default: false },
+    description: { type: 'string' },
+  },
+};
+
+
+export const ConfigRoot = {
```

Comment:
Why removed the type?

```suggestion
export const ConfigRoot: NodeType = {
```